### PR TITLE
Fix some eslint warnings in YoastSEO.js

### LINKED ISSUE #1847: ESLint rule no-shadow
In PR https://github.com/Yoast/YoastSEO.js/pull/1846 the ESLint errors are temporarily put back to warnings.

The job for this issue is to adapt the code so that the `no-shadow` rule overrides can go back to error. See `.eslintrc` and `.eslintrc-tests`.

### LINKED ISSUE #1848: ESLint rule require-jsdoc
In PR https://github.com/Yoast/YoastSEO.js/pull/1846 the ESLint errors are temporarily put back to warnings.

The job for this issue is to adapt the code so that the `require-jsdoc` rule overrides can go back to error. See `.eslintrc` and `.eslintrc-tests`.

### PR DESCRIPTION
**Pull request by nataliashitova**
## Summary

This PR can be summarized in the following changelog entry:

* [no-user-facing-changes] Fixes a number of `require-jsdoc` and `no-shadow` eslint warnings.
* Fixes a but where a Flesch reading ease score of exactly 90 would incorrectly trigger bad feedback.

## Relevant technical choices:

* Does not fix warnings in `app.js`and `appSpec.js` because they will be deprecated anyway #2045

## Test instructions

This PR can be tested by following these steps:

* Run `grunt check` and check if nothing fails.
* Run `yarn test` and check if nothing fails.
* Since this PR included some refactoring of the **Flesch Reading Ease assessment**, please check based on this information whether you can get all possible feedback strings:
  * **Red 3** Flesch Reading Ease: The copy scores 0 in the test, which is considered very difficult to read. Try to make shorter sentences, using less difficult words to improve readability.
  * **Red 3** Flesch Reading Ease: The copy scores 45 in the test, which is considered difficult to read. Try to make shorter sentences, using less difficult words to improve readability.
  * **Orange 6** Flesch Reading Ease: The copy scores 55 in the test, which is considered fairly difficult to read. Try to make shorter sentences to improve readability.
  * **Green 9** Flesch Reading Ease: The copy scores 63.9 in the test, which is considered ok to read. Good job!
  * **Green 9** Flesch Reading Ease: The copy scores X in the test, which is considered fairly easy to read. Good job!
  * **Green 9** Flesch Reading Ease: The copy scores X in the test, which is considered easy to read. Good job!
  * **Green 9** Flesch Reading Ease: The copy scores X in the test, which is considered very easy to read. Good job!
  * Check if the assessment title link works
  * Check if the assessment CTA link appears for all Orange and Red results and works.
  * Check if the language-specific configurations are still applied. For instance, try to get a Russian paper that scores between 40 and 50 and make sure to get an orange bullet (the same score for an English paper should result in a red bullet).

Fixes #1847 Fixes #1848

### CODE REVIEW

**Comment by manuelaugustin** [comment]
CR 🏗

Down from 84 warnings to 49, nice job! 🎉

I had some small remarks, most of which I've added as suggestions.

In terms of acceptance testing I think it might be a good idea to add a quick manual test of the Flesch reading ease assessment to the testing instructions, since there has been a bit more internal refactoring to that assessment than just changing variable names.

File: src/researches/german/passiveVoice/determinePassives.js [hunk 1]
```diff
@@ -1,9 +1,9 @@
 import arrayToRegex from "../../../stringProcessing/createRegexFromArray.js";
-var auxiliaries = require( "./auxiliaries.js" )().allAuxiliaries;
+var germanAuxiliaries = require( "./auxiliaries.js" )().allAuxiliaries;
```
[the remark this answers is not in the document]
Comment (nataliashitova) [reply src/researches/german/passiveVoice/determinePassives.js]: Fixed!

File: src/assessments/readability/fleschReadingEaseAssessment.js [hunk 2]
```diff
@@ -65,76 +68,78 @@ class FleschReadingEaseAssessment extends Assessment {
...
+			note = i18n.dgettext( "js-text-analysis", "Try to make shorter sentences, using less difficult words to improve readability" );
+		} else {
+			score = this._config.scores.veryDifficult;
+			feedback = i18n.dgettext( "js-text-analysis", "very difficult" );
+			note = i18n.dgettext( "js-text-analysis", "Try to make shorter sentences, using less difficult words to improve readability" );
 		}
 
-		if ( inRange( this.fleschReadingResult, this._config.borders.okay, this._config.borders.fairlyEasy ) ) {
-			const feedback = i18n.dgettext( "js-text-analysis", "ok" );
+		if ( score >=  this._config.scores.okay ) {
```
[the remark this answers is not in the document]
Comment (nataliashitova) [reply src/assessments/readability/fleschReadingEaseAssessment.js]: Agreed!

**Comment by manuelaugustin** [comment]
CR & acceptance 👍

**Pull request merged**